### PR TITLE
Improve the Spotify Integration code

### PR DESCRIPTION
**Pull request by amCap1712**
Now that we are moving to new schema structure which supports integrating multiple services, it makes sense to use this opportunity to rewrite existing code to promote reuse between different services and make adding new integrations easier.

### CODE REVIEW

**Comment by amCap1712** [comment]
@MonkeyDo I have applied the relevant commits from #880 to this PR. I also made some additional changes on the frontend handling of spotify permissions as an array as necessary.

File: listenbrainz/webserver/static/js/src/SpotifyPlayer.tsx [hunk 1]
```diff
@@ -212,7 +212,7 @@ export default class SpotifyPlayer
         }
       })
       .catch((error) => {
-        handleError(error);
+        handleError(error.message);
```
Comment (MonkeyDo) [inline listenbrainz/webserver/static/js/src/SpotifyPlayer.tsx:215]: Looking at the implementation of handleError, this shouldn't be needed:
https://github.com/metabrainz/listenbrainz-server/blob/master/listenbrainz/webserver/static/js/src/BrainzPlayer.tsx#L164-L178

**Review by MonkeyDo — COMMENTED** [review]
The Javascript bits look good !

File: listenbrainz/webserver/static/js/src/SpotifyPlayer.tsx (hunk 1 above)
Comment (amCap1712) [reply listenbrainz/webserver/static/js/src/SpotifyPlayer.tsx:215]: Will fix.

**Comment by amCap1712** [comment]
Cool. Let's close #880 in favour this PR then.

**Pull request merged**